app/models/king.rb

```diff
@@ -1,12 +1,90 @@
 class King < Piece
-  # This method does not yet consider castling.
   def valid_move?(new_x, new_y)
     return false if off_board?(new_x, new_y)
     return false if current_square?(new_x, new_y)
 
     x_offset, y_offset = movement_by_axis(new_x, new_y)
-    return false unless (-1..1).cover?(x_offset) && (-1..1).cover?(y_offset)
+    return true if (-1..1).cover?(x_offset) && (-1..1).cover?(y_offset)
 
-    true
+    valid_castling?(new_x, new_y)
+  end
+
+  # An addition to the move_to! method found within the Piece model.
+  # This determines which rook is to be castled and moves it to the
+  # appropriate square.
+  def move_to!(destination_x, destination_y)
+    return false unless valid_move?(destination_x, destination_y)
+    castling = (x_coord - destination_x).abs == 2
+
+    # Call the move_to! method within the Piece model and move the king.
+    super
+
+    # If all of the requirements are met for valid_move? AND the intention is to castle;
+    # place the rook in its appropriate position to complete the castling move.
+    if castling
+      rook_origin_file = (destination_x == 3) ? 1 : 8
+      rook_destination_file = (destination_x == 3) ? 4 : 6
+      castled_rook = game.piece_at(rook_origin_file, destination_y)
+      castled_rook.update_attributes(x_coord: rook_destination_file,
+                                     y_coord: destination_y,
+                                     moved: true)
+    end
+  end
+
+  private
+
+  # Determine if castling is allowed.
+  def valid_castling?(new_x, new_y)
+    # The right to castle has been lost if the king has moved.
+    return false if moved?
+
+    # Castling is permitted if the king and the chosen rook are on the player's first rank.
+    # Since the king hasn't moved, then y_coord will be the same as the player's first rank.
+    return false unless y_coord == new_y
+
+    # Castling is not permitted unless the king moves two spaces along the same rank.
+    return false unless new_x == 3 || new_x == 7
+
+    # Castling is not permitted if the rook has moved, or if there are any obstructions.
+    return false unless rook_requirements_met?(new_x, new_y)
+
+    # Allows Castling if the squares that the king traverses are not under attack.
+    !king_traversal_under_attack?(new_x, new_y)
+  end
+
+  # Castling helper method.
+  def rook_requirements_met?(new_x, new_y)
+    # Set the file for the rook to 1 if the player's intent is to Castle queen-side,
+    # otherwise it will be set to eight. File is a chess term that's analogous to x_coord.
+    rook_file = new_x == 3 ? 1 : 8
```

Comment:
I think you can remove stuff about `file` unless it's necessary. why not just replace`file` with `x_coord`?